src/main/java/io/kestra/plugin/nats/Request.java

```diff
@@ -0,0 +1,198 @@
+package io.kestra.plugin.nats;
+
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.runners.RunContext;
+import io.nats.client.Connection;
+import io.nats.client.Message;
+import io.nats.client.impl.Headers;
+import io.nats.client.impl.NatsMessage;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.*;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Send a request to a NATS subject and wait for a reply."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Send a request to the subject and wait for the reply (using username/password authentication).",
+            full = true,
+            code = """
+                id: nats_request_reply
+                namespace: company.team
+
+                tasks:
+                  - id: request
+                    type: io.kestra.plugin.nats.Request
+                    url: nats://localhost:4222
+                    username: nats_user
+                    password: nats_password
+                    subject: "greet.bob"
+                    from:
+                      headers:
+                        someHeaderKey: someHeaderValue
+                      data: "Hello from Kestra!"
+                    requestTimeout: 2000
+                """
+        )
+    }
+)
+public class Request extends NatsConnection implements RunnableTask<Request.Output> {
+    @Schema(
+        title = "Subject to send the request to"
+    )
+    @PluginProperty(dynamic = true)
+    @NotBlank
+    @NotNull
+    private String subject;
```

Comment:
```suggestion
    @NotNull
    private Property<String> subject;
```

We have a new way, more powerful, to deal with dynamic properties.
See here for more detail: https://kestra.io/docs/plugin-developer-guide/task#properties